server/tests/api_v4/authenticated/test_events_listen.py

```diff
@@ -41,6 +43,18 @@ async def test_events_listen_ok(
 
 
 # TODO: test for each event type
-# TODO: test keepalive
 # TODO: test `Last-Event-ID`
 # TODO: test connection gets closed due to SseAPiEventsListenBadOutcome
+
+
+@pytest.mark.timeout(2)
+async def test_keep_alive(minimalorg: MinimalorgRpcClients, backend: Backend) -> None:
+    # Reduce keepalive to 0.1s to speed up the test
+    backend.config.sse_keepalive = 0.1
+
+    async with minimalorg.alice.raw_sse_connection() as raw_sse_stream:
+        async for line in raw_sse_stream.aiter_lines():
+            line = line.rstrip("\n")
+
+            if line == ":keepalive":
+                break
```

Comment:
Should add a boolean `got_keepalive` that is checked at the end of the test
Otherwise, the test could appear passing the async with or async for skip it body for whatever reason

Reply:
Indeed